google/cloud/storage/tests/universe_domain_integration_test.cc

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/client.h"
+#include "google/cloud/storage/options.h"
+#include "google/cloud/storage/testing/storage_integration_test.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include "google/cloud/universe_domain.h"
+#include "google/cloud/universe_domain_options.h"
+#include "google/cloud/version.h"
+#include <gmock/gmock.h>
+#include <fstream>
+#include <string>
+#include <utility>
+
+namespace google {
+namespace cloud {
+namespace storage {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::google::cloud::internal::GetEnv;
+
+class UniverseDomainIntegrationTest
+    : public google::cloud::storage::testing::StorageIntegrationTest {
+ protected:
+  void SetUp() override {
+    bucket_name_ = MakeRandomBucketName();
+    object_name_ = MakeRandomObjectName();
+  }
+
+  std::string const& bucket_name() const { return bucket_name_; }
+  std::string const& object_name() const { return object_name_; }
+
+ private:
+  std::string bucket_name_;
+  std::string object_name_;
+};
+
+auto TestOptions() {
+  auto sa_key_file = GetEnv("UD_SA_KEY_FILE").value_or("");
```

Comment:
How are we setting the value of these environment variables?

Reply:
Actually your comment made me realize I had to actually pass these to bazel test. Updated the code.